source/agora/network/Manager.d

```diff
@@ -1119,6 +1123,8 @@ public class NetworkManager
         settings.httpClientSettings = new HTTPClientSettings;
         settings.httpClientSettings.connectTimeout = timeout;
         settings.httpClientSettings.readTimeout = timeout;
+        if (this.proxy_url != string.init)
```

Comment:
```suggestion
        if (this.proxy_url.length)
```

Little piece of D knowledge: `if (myString)` is `true` for `""`, because it's equivalent to `if (myString.ptr !is null)`. Since `string.init` is `null` (and not `""`), if we ended up with an empty string here, the check would pass.

And now that I wrote this, I realize that `URL.parse` should probably be called in the config parsing, so we don't even deal with strings here...

Reply:
> And now that I wrote this, I realize that URL.parse should probably be called in the config parsing, so we don't even deal with strings here...

Yeah makes sense, will update

Reply:
Updated